test/test_shared.cpp

```diff
@@ -2381,6 +2381,26 @@ TEST(Shared_MixedWithNonShared)
 #endif
 }
 
+
+#if REALM_ENABLE_ENCRYPTION
+// verify that even though different threads share the same encrypted pages,
+// a thread will not get access without the key.
+TEST(Shared_EncryptionKeyCheck)
```

Comment:
Should we have a more comprehensive test? I think all 3 different paths should be tested.